From c02ab25592308fa6fc442ec6330d1128fec98385 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 22 May 2017 19:08:41 +0800
Subject: [PATCH 1817/5242] MLK-14963 PXP: enable pxp interrupt after soft
 reset

commit  624f4cdc154eb9deb6bc6a36e10417bf6fe414f4 from
https://source.codeaurora.org/external/imx/linux-imx.git

After soft reset, the irq register value will be zero, so we need set it to enable
all pxp interrupts.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index 66a4daa..ac57291 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -7420,12 +7420,6 @@ static int pxp_probe(struct platform_device *pdev)
 
 	pxp_soft_reset(pxp);
 	pxp_writel(0x0, HW_PXP_CTRL);
-
-	/* Initialize PXP Interrupt */
-	err = pxp_init_interrupt(pdev);
-	if (err < 0)
-		goto exit;
-
 	/* Initialize DMA engine */
 	err = pxp_dma_init(pxp);
 	if (err < 0)
@@ -7433,6 +7427,12 @@ static int pxp_probe(struct platform_device *pdev)
 
 	pxp_clk_enable(pxp);
 	pxp_soft_reset(pxp);
+
+	/* Initialize PXP Interrupt */
+	err = pxp_init_interrupt(pdev);
+	if (err < 0)
+		goto exit;
+
 	if (pxp->devdata && pxp->devdata->pxp_data_path_config)
 		pxp->devdata->pxp_data_path_config(pxp);
 
-- 
1.7.9.5

